Overhaul first-timer-support page (#1870)
adoptium/adoptium.net: content/asciidoc-pages/contributing/index.adoc

== Submitting a contribution to Adoptium

After signing the ECA, you can propose contributions by sending pull requests (PRs) through GitHub.

**Note:** Assume we are contributing to the `aqa-tests` repo

If this is your first time contributing to the project, fork the repo by clicking on the Fork button in the top-right corner of the git repo page. This creates a copy of the repo under your GitHub account: https://github.com/<YourGitUserName>/aqa-tests.git

Git clone the your repo:

[source, bash]
----
git clone https://github.com/<YourGitUserName>/aqa-tests.git
----

Create a new branch to work on:

[source, bash]
----
cd aqa-tests
git checkout -b my_new_branch
----

- Add https://github.com/adoptium/aqa-tests.git as your upstream:

[source, bash]
----
git remote add upstream https://github.com/adoptium/aqa-tests.git
----

Before you start working on the issue, plese make sure the local branch is up to date:

[source, bash]
----
git fetch upstream
git rebase upstream/master
----

Once you are done with your work, track your changes and commit.

[source, bash]
----
git add .
git commit -s -m "message about this PR"
----

- Commit message example:

> [source, bash]

Reviewer: ```suggestion
[source]
```

> = Contributing to Adoptium

Reviewer: ```suggestion
= Contributing to Adoptium(R)
```

One of the key objectives of Adoptium is to make it easier for developers and organizations to access and consume OpenJDK builds. It provides prebuilt binaries that are thoroughly tested, ensuring stability and compatibility. These binaries are available for free, making it convenient for developers to adopt and use OpenJDK in their projects without any licensing restrictions.

Adoptium also offers long-term support (LTS) builds, which are specifically designed for enterprises and organizations that require a stable and reliable Java runtime environment. The LTS builds receive regular updates and security patches, ensuring the longevity and maintenance of the Java platform.

Overall, Adoptium aims to provide a trustworthy and accessible distribution of OpenJDK, promoting transparency, collaboration, and community-driven innovation within the Java ecosystem.

- https://github.com/adoptium

> == Developer resources

Reviewer: I'm guessing this section was copied over from the markdown file. Please remove it as it's not generic

The AQAvit project maintains the following source code repositories:

- https://github.com/adoptium/temurin-build
- https://github.com/adoptium/aqa-tests
- https://github.com/adoptium/installer
- https://github.com/adoptium/adoptium-support
- https://github.com/adoptium/containers
- https://github.com/adoptium/adoptium.net
- And many https://github.com/adoptium/[others]

== Eclipse Contributor Agreement

Before your contribution can be accepted by the project team contributors must electronically sign the Eclipse Contributor Agreement (ECA).

- http://www.eclipse.org/legal/ECA.php

Commits that are provided by non-committers must have a Signed-off-by field in the footer indicating that the author is aware of the terms by which the contribution has been provided to the project. The non-committer must additionally have an Eclipse Foundation account and must have a signed Eclipse Contributor Agreement (ECA) on file.

For more information, please see the Eclipse Committer Handbook and Adoptium documentation:

- https://www.eclipse.org/projects/handbook/#resources-commit

> - https://adoptium.net/docs/eca-sign-off

Reviewer: This should use a relative link e.g /docs/eca-sign-off

== Submitting a contribution to Adoptium

> After signing ECA, you can propose contributions by sending pull requests (PRs) through GitHub.

Reviewer: ```suggestion
After signing the ECA, you can propose contributions by sending pull requests (PRs) through GitHub.
```

> **Note:** Assume we are `aqa-tests` repo

Reviewer: ```suggestion
**Note:** Assume we are contributing to the `aqa-tests` repo
```

----

Before you start working on the issue, plese make sure the local branch is up to date:

[source, bash]
----
git fetch upstream
git rebase upstream/master
----

Once you are done with your work, track your changes and commit.

[source, bash]
----
git add .
git commit -s -m "message about this PR"
----

- Commit message example:

[source, bash]
----
Update jtreg in openjdk test
This patch updates jtreg version from 5.1 to 6.1.
Fixes: #1234
Signed-off-by: Full Name <email>
----

- The first line is the PR title. It should describe the change made. Please keep it short and simple.
- The body should include detailed information about your PR. You may want to include designs, rationale, and a brief explanation of what you have changed. Please keep it concise.
- When a commit has related issues, please use the Git keywords to automatically close or relate to the issues. https://help.github.com/articles/closing-issues-using-keywords/
- Please sign off on your commit in the footer. This can be automatically added to your commit by passing -s to git commit (see above example).

Push the change into your Git repo:

[source, bash]
----

> git push -f origin my_new_branch

Reviewer: ```suggestion
git push origin my_new_branch
```

I'd rather not recommend force pushing

----

> - You can skip -f in git push if you've never pushed your branch before.

Reviewer: ```suggestion
- You can use -f in git push if need to overwrite your git history. Please use this flag with caution.
```

> We would like to encourage you to open a pull request early and use `Create draft pull request` option. This allows others to check the PR, get early feedback, and helps create a better end product.

Reviewer: ```suggestion
We would like to encourage you to open a pull request early and use the `Create draft pull request` option. This allows others to check the PR, get early feedback, and helps create a better end product.
```

> If you have been given access to run test jobs in our Jenkins server, run https://ci.adoptium.net/job/Grinder/[Grinder] to validate your PR. Your can find recordings about how to use Grinder in AQA Lightning Talk Series: https://github.com/eclipse-openj9/openj9/wiki/AQA-Lightning-Talk-Series. If you do not have access, the reviewers of your PR will run some tests. Reviewers may ask you to run extra tests depending on what changes you have made in your PR.

Reviewer: Please remove - specific to AQA-test

> Ensure all related Grinder jobs pass and provide the Grinder links in the PR comment. Your changes must also pass the auto PR builds that will be applied to your pull request.

Reviewer: Please remove - specific to AQA test
